sql/sql_table.cc

```diff
@@ -6418,22 +6419,26 @@ static bool fill_alter_inplace_info(THD *thd,
 
   new_field_it.init(alter_info->create_list);
   while ((new_field= new_field_it++))
-  {
-    if (! new_field->field)
-    {
-      // Field is not present in old version of table and therefore was added.
-      if (new_field->vcol_info)
-        if (new_field->stored_in_db())
-          ha_alter_info->handler_flags|=
-            Alter_inplace_info::ADD_STORED_GENERATED_COLUMN;
-        else
-          ha_alter_info->handler_flags|=
-            Alter_inplace_info::ADD_VIRTUAL_COLUMN;
-      else
-        ha_alter_info->handler_flags|=
-          Alter_inplace_info::ADD_STORED_BASE_COLUMN;
-    }
-  }
+	{
+		if (! new_field->field)
+		{
+			// Field is not present in old version of table and therefore was added.
+			if (new_field->vcol_info)
+				if (new_field->stored_in_db())
+					ha_alter_info->handler_flags|=
+					Alter_inplace_info::ADD_STORED_GENERATED_COLUMN;
+				else
+					ha_alter_info->handler_flags|=
+					Alter_inplace_info::ADD_VIRTUAL_COLUMN;
+			else {
+				if (new_field->has_default_expression() && !new_field->has_default_now_unireg_check()) {
+					has_expr_default = true;
+				}
+				ha_alter_info->handler_flags|=
+					Alter_inplace_info::ADD_STORED_BASE_COLUMN;
+			}
+		}
+	}
```

Comment:
Please try to follow the existing indentation rules and avoid changes to white space. InnoDB follows different indentation rules than the rest of the server.